lib/Data/PhoneNumber/FFI.chs

```diff
@@ -46,6 +47,8 @@ data PhoneNumberUtil = PhoneNumberUtil { unPhoneNumberUtil :: Ptr PhoneNumberUti
 
 {# enum PhoneNumberFormat as PhoneNumberFormat {underscoreToCase} deriving (Eq, Show) #}
 
+{# enum ErrorType as PhoneNumberParseError {underscoreToCase} deriving (Eq, Show) #}
```

Comment:
Is this some FFI based declaration of a Haskell datatype? From what this looks like: 
- You're taking the enum literal from C as is 
- PhoneNumberParseError is a datatype with a _single_ dataconstructor that accepts the camelcased C enum type 

Are my assumptions correct?

Reply:
This is an import of a C enum into a Haskell sum type.
What it does: it takes the C enum `ErrorType` and converts every upper-snake-case value in the enum to a data constructor of a new Haskell `PhoneNumberParseError` type.

For `ErrorType = enum {NO_PARSING_ERROR, TOO_LONG_NSN,...}` it gives `data PhoneNumberParseError =  NoParsingError | TooLongNsn | ... deriving (Eq, Show)`

Reply:
Okay, and this seems to be a standard way to do this on the FFI with C. Correct?

Reply:
It is, using c2hs.

Reply:
I have pushed a new commit that hides the NoParsingError that occurred when parsing succeeded. It was already never constructed, but I now removed the constructor itself.

Reply:
Okay, cool. Yeah this seems like a reasonable solution for me, and also something that might require little to no intervention in case more enums are added to ErrorType in the underlying lib. 

LGTM.